plugins/modules/sonic_dcbx.py

```diff
@@ -0,0 +1,466 @@
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+
+# Using "merged" state for DCBx interface configuration
+#
+# Before State:
+# -------------
+#
+# sonic# show running-configuration | grep dcbx
+# sonic#
+
+  - name: Modify DCBX configurations
+    dellemc.enterprise_sonic.sonic_dcbx:
+      config:
+        global:
+            enabled: true
+      state: merged
+
+# After State:
+# ------------
+# sonic# show running-configuration | grep dcbx
+# !
+# dcbx enable
+# !
+
+# Using "merged" state for DCBx interface configuration
+#
+# Before State:
+# -------------
+# sonic# show running-configuration interface Ethernet 0
+# !
+# interface Ethernet0
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+#  no dcbx enable
+#  no dcbx tlv-select pfc
+#  no dcbx tlv-select ets-conf
+#  no dcbx tlv-select ets-reco
+
+
+  - name: Delete and set default DCBx Interface configurations
+    dellemc.enterprise_sonic.sonic_dcbx:
+      config:
+        interfaces:
+          - name: Ethernet0
+            enabled: False
+            pfc_tlv_enabled: False
+            ets_configuration_tlv_enabled: False
+            ets_recommendation_tlv_enabled: False
+      state: deleted
+
+# After State:
+# ------------
+# sonic# show running-configuration interface Ethernet0
+# !
+# interface Ethernet0
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+
+# Using "merged" state for DCBx interface configuration
+#
+# Before State:
+# -------------
+# sonic# show running-configuration interface Ethernet 24
+# !
+# interface Ethernet24
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+#  no dcbx enable
+#  no dcbx tlv-select pfc
+#  no dcbx tlv-select ets-conf
+#  no dcbx tlv-select ets-reco
+
+
+  - name: Delete and set default DCBx Interface configurations
+    dellemc.enterprise_sonic.sonic_dcbx:
+      config:
+        interfaces:
+          - name: Ethernet0
+            pfc_tlv_enabled: False
+      state: deleted
+
+# After State:
+# ------------
+# sonic# show running-configuration interface Etherneti 24
+# !
+# interface Ethernet24
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+#  no dcbx enable
+#  no dcbx tlv-select ets-conf
+#  no dcbx tlv-select ets-reco
+
+# Using "replaced" state for DCBx interface configuration
+#
+# Before State:
+# -------------
+# sonic# show running-configuration interface Ethernet9
+# !
+# interface Ethernet0
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+
+  - name: Modify DCBx Interface configurations
+    dellemc.enterprise_sonic.sonic_dcbx:
+      config:
+        -interfaces:
+          - name: Ethernet0
+            enabled: False
+      state: replaced
+
+# After State:
+# ------------
+# sonic# show running-configuration interface Ethernet0
+# !
+# interface Ethernet0
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+#  no dcbx enable
+
+# Using "replaced" state for DCBx global configuration
```

Comment:
This example is not necessary. (Replaced state is not useful for a global configuration with only one option. It is the same as "merged" for this case.)

This example should be deleted to avoid confusing customers who read it.

Reply:
deleted.